File: informatika_ismeretek/kozep/2019_okt/java/Cbradio.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Cbradio {

    public static void main(String[] args) throws IOException {
        var lines = Files.readAllLines(Path.of("cb.txt"));
        var bejegyzesek = new ArrayList<Bejegyzes>();

        for(var i = 1; i < lines.size(); i++) {
            bejegyzesek.add(new Bejegyzes(lines.get(i)));
        }

        System.out.println("3. Feladat: Bejegyzések száma: " + bejegyzesek.size());

        var voltE4AdastIndito = false;
        for(var bejegyzes : bejegyzesek) {
            if(bejegyzes.adasok == 4) {
                voltE4AdastIndito = true;
                break;
            }
        }

        System.out.println("4. Feladat: " + (voltE4AdastIndito ? "Volt" : "Nem volt") + " 4 adást indító sofőr");

        var bekertNev = System.console().readLine("5. Feladat: Írj be egy nevet: ");
        var bekertHasznalatok = 0;

        for(var bejegyzes : bejegyzesek) {
            if(bejegyzes.nev.equals(bekertNev)) {
                bekertHasznalatok += bejegyzes.adasok;
            }
        }

        if(bekertHasznalatok > 0) {
            System.out.println(bekertNev + " " + bekertHasznalatok + "x használta a rádiót");
        }else{
            System.out.println("Nincs ilyen nevű sofőr!");
        }

        try(var file = new PrintWriter("cb2.txt")){
            file.println("Kezdes;Nev;AdasDb");

            for(var bejegyzes : bejegyzesek) {
                file.println(atszamolPercre(bejegyzes.ora, bejegyzes.perc) + ";" + bejegyzes.nev + ";" + bejegyzes.adasok);
            }
        }

        var egyediSoforok = new HashSet<String>();
        for(var bejegyzes : bejegyzesek) {
            egyediSoforok.add(bejegyzes.nev);
        }

        System.out.println("8. Feladat: Sofőrök száma: " + egyediSoforok.size());

        var soforokAdasszamokkal = new HashMap<String, Integer>();
        for(var bejegyzes : bejegyzesek) {
            var soforNeve = bejegyzes.nev;

            soforokAdasszamokkal.put(soforNeve, soforokAdasszamokkal.getOrDefault(soforNeve, 0) + bejegyzes.adasok);
        }

        var legtobbAdasBejegyzes = soforokAdasszamokkal.entrySet().iterator().next();
        for(var bejegyzes : soforokAdasszamokkal.entrySet()) {
            if(bejegyzes.getValue() > legtobbAdasBejegyzes.getValue()) {
                legtobbAdasBejegyzes = bejegyzes;
            }
        }

        System.out.println("9. Feladat: Legtöbb adást indító sofőr: " + legtobbAdasBejegyzes.getKey() + ", adások: " + legtobbAdasBejegyzes.getValue());
    }

    public static int atszamolPercre(int ora, int perc) {
        return ora * 60 + perc;
    }
}

File: informatika_ismeretek/kozep/2019_okt/c#/Cbradio.cs
using System;
using System.IO;
using System.Collections.Generic;

public class Cbradio {

    public static void Main(string[] args) {
        var lines = File.ReadAllLines("cb.txt");
        var bejegyzesek = new List<Bejegyzes>();

        for(var i = 1; i < lines.Length; i++) {
            bejegyzesek.Add(new Bejegyzes(lines[i]));
        }

        Console.WriteLine("3. Feladat: Bejegyzések száma: " + bejegyzesek.Count);

        var voltE4AdastIndito = false;
        foreach(var bejegyzes in bejegyzesek) {
            if(bejegyzes.adasok == 4) {
                voltE4AdastIndito = true;
                break;
            }
        }

        Console.WriteLine("4. Feladat: " + (voltE4AdastIndito ? "Volt" : "Nem volt") + " 4 adást indító sofőr");
        Console.WriteLine("5. Feladat: Írj be egy nevet");

        var bekertNev = Console.ReadLine();
        var bekertHasznalatok = 0;

        foreach(var bejegyzes in bejegyzesek) {
            if(bejegyzes.nev == bekertNev) {
                bekertHasznalatok += bejegyzes.adasok;
            }
        }

        if(bekertHasznalatok > 0) {
            Console.WriteLine(bekertNev + " " + bekertHasznalatok + "x használta a rádiót");
        }else{
            Console.WriteLine("Nincs ilyen nevű sofőr!");
        }

        using(var file = new StreamWriter("cb2.txt")){
            file.WriteLine("Kezdes;Nev;AdasDb");

            foreach(var bejegyzes in bejegyzesek) {
                file.WriteLine(atszamolPercre(bejegyzes.ora, bejegyzes.perc) + ";" + bejegyzes.nev + ";" + bejegyzes.adasok);
            }
        }

        var egyediSoforok = new HashSet<string>();
        foreach(var bejegyzes in bejegyzesek) {
            egyediSoforok.Add(bejegyzes.nev);
        }

        Console.WriteLine("8. Feladat: Sofőrök száma: " + egyediSoforok.Count);

        var soforokAdasszamokkal = new Dictionary<string, int>();
        foreach(var bejegyzes in bejegyzesek) {
            var soforNeve = bejegyzes.nev;
            var oldValue = 0;

            soforokAdasszamokkal.TryGetValue(soforNeve, out oldValue);
            soforokAdasszamokkal.Add(soforNeve, oldValue + bejegyzes.adasok);
        }

        var legtobbAdasBejegyzes = soforokAdasszamokkal.GetEnumerator().Current;
        foreach(var bejegyzes in soforokAdasszamokkal) {
            if(bejegyzes.Value > legtobbAdasBejegyzes.Value) {
                legtobbAdasBejegyzes = bejegyzes;
            }
        }

        Console.WriteLine("9. Feladat: Legtöbb adást indító sofőr: " + legtobbAdasBejegyzes.Key + ", adások: " + legtobbAdasBejegyzes.Value);
    }

    public static int atszamolPercre(int ora, int perc) {
        return ora * 60 + perc;
    }
}